builtInFunctions/changeOwnerAddress.go

```diff
@@ -55,9 +65,26 @@ func (c *changeOwnerAddress) ProcessBuiltinFunction(
 		return nil, ErrNotEnoughGas
 	}
 	gasRemaining := computeGasRemaining(acntSnd, vmInput.GasProvided, c.gasCost)
+
+	vmOutput := &vmcommon.VMOutput{ReturnCode: vmcommon.Ok, GasRemaining: gasRemaining}
+	isCrossShardCallThroughASmartContract := check.IfNil(acntDst) && vmcommon.IsSmartContractAddress(vmInput.CallerAddr)
```

Comment:
I think the if check and the call to addOutputTransferToVMOutput can be added after L85, in which case, the if condition simplifies by one check. You can even extract it in a new function to simplify how the code looks. L69 can remain, the code looks better with it

Reply:
refactored.